src/main/java/com/nha/abdm/wrapper/hip/hrp/GatewayCallbackController.java

```diff
@@ -90,15 +93,22 @@ public void confirmCall(@RequestBody ConfirmResponse confirmResponse)
    * @param linkOnInitResponse Response from ABDM gateway after auth/init which has transactionId.
    */
   @PostMapping({"/v0.5/users/auth/on-init"})
-  public void onAuthInitCall(@RequestBody LinkOnInitResponse linkOnInitResponse) {
+  public ErrorResponse onAuthInitCall(@RequestBody LinkOnInitResponse linkOnInitResponse)
+      throws IllegalDataStateException {
     if (linkOnInitResponse != null && linkOnInitResponse.getError() == null) {
-      log.info(linkOnInitResponse.toString());
-      this.workflowManager.initiateAuthConfirmDemographics(linkOnInitResponse);
+      log.debug(linkOnInitResponse.toString());
+      this.workflowManager.initiateAuthConfirm(linkOnInitResponse);
     } else if (linkOnInitResponse.getError() != null) {
-      log.info("got Error in OnInitRequest callback: " + linkOnInitResponse.getError().toString());
+      updateRequestError(
+          linkOnInitResponse.getResp().getRequestId(),
+          "onAuthInitCall",
+          linkOnInitResponse.getError().getMessage());
     } else {
-      log.error("got Error in OnInitRequest callback");
+      String error = "Got Error in OnInitRequest callback: gateway response was null";
+      updateRequestError(linkOnInitResponse.getResp().getRequestId(), "onAuthInitCall", error);
+      return ErrorResponse.builder().code(1000).message(error).build();
```

Comment:
Whats the importance of error code `1000` here? The error codes probably could be stored in a central place.

Reply:
This is the code shown in ABDM swagger contracts. I am also not sure. @VenuAjitesh can you please provide some details if possible.